File: leetcode/lc-us/cpp/1822.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int arraySign(vector<int> &nums) {
        int count = 0;
        for (int n: nums) {
            if (n == 0) return 0;
            if (n < 0) count++;
        }
        return count & 1 ? -1 : 1;
    }
};

File: leetcode/lc-us/rust/1822.rs
impl Solution {
    pub fn array_sign(nums: Vec<i32>) -> i32 {
        let mut cnt_minus_1 = 0;
        for x in nums {
            if x == 0 {
                return 0;
            } else if x < 0 {
                cnt_minus_1 += 1;
            }
        }
        return if cnt_minus_1 & 1 == 0 { 1 } else { -1 };
    }
}